src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/RssCheckingActor.java

```diff
@@ -32,17 +31,13 @@ public static Props props(ActorRef postStoringActor, SyndFeedProducer syndFeedFa
     @SuppressWarnings("unchecked")
     private void executeAction(ActorRef postStoringActor, SyndFeedProducer syndFeedFactory,
                                RssLink rssLink) {
-        try {
-            SyndFeed feed = syndFeedFactory.createFor(rssLink.getUrl());
-            List<SyndEntry> posts = feed.getEntries();
-            posts.forEach(post -> {
+        syndFeedFactory.createFor(rssLink.getUrl()).ifPresent(feed -> {
```

Comment:
I did a little cosmetic change here, however it doesn't mean I disagree with you :). This case has just an aesthetic reason to be changed (in both case it doesn't get rid of lambdas totally). It might get back to classical branching (`if` usage) in case project will have a rule for not using `isPresent` with functional expression to prevent such mistakes in future.